src/util/LazyJsonParser.h

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+
+#ifndef QLEVER_LAZY_JSON_PARSER_H_
+#define QLEVER_LAZY_JSON_PARSER_H_
+
+#include "util/json.h"
+
+namespace ad_utility {
+/*
+ * A simple parser for split up JSON-data with known structure.
+ *
+ * Given the path to an array containing the majority of the given JSON-object,
+ * the Parser will yield chunks of the object separated after completed elements
+ * in the arrayPath or after reading the entire object itself.
+ */
+class LazyJsonParser {
+ public:
+  explicit LazyJsonParser(std::vector<std::string> arrayPath);
+
+  // TODO<unexenu> this should take a generator as argument instead
+  static std::string parse(const std::string& s,
+                           const std::vector<std::string>& arrayPath) {
+    LazyJsonParser p(arrayPath);
+    return p.parseChunk(s);
+  }
```

Comment:
We also need some safety measure to make sure that we cannot haven an denial of service attack with ill-formed JSON
(basically some thresholds like "after 1 Megabyte we must be in the array path, each entry in the arrayPath must be at most 1 MB  and the same goes for everything after the array path.

This should be relatively easy to track.

Reply:
I've added a simple check for that in the latest commit.

While it's possible to compute this precisely, the current solution compares the first/last element in the arrayPath together with the part before/after the arrayPath against the threshold.
All other elements are checked one by one.

This leads to the first/last element of arrayPath being more restricted in size than the other elements.

Reply:
This mechanism now looks good to me.